### ISSUE TITLE
error when rendering multixy plot with nofill option enabled

### ISSUE THREAD

**Issue by jbrestel** [elsiklab/multibigwig#37]
![2023-10-27 - 15-52-34 - PlasmoDB -- JBrowse](https://github.com/elsiklab/multibigwig/assets/39096257/dabb1426-7b00-4e43-812b-6615d7c6b9cb)

The image above shows individual XY tracks and a multibw / combined track.  These should match but several places where the multibw track adds lines (especially around intron junctions / gaps in coverage)

**Comment by cmdcolin** [comment]
the plugin does try to draw a line from zero to try to make the lines look more natural (it is not just doing nofill rendering but instead tries to make continuous line style renderings) but particular bigwigs which I call sparse bigwigs can challenge this. seeing same thing in jbrowse 2. could probably remove logic that draws the line to zero but I kind of feel like if the problem could be solved, that the lines to zero are helpful so it doesn't just look like a line starting from "nowhere"

**Comment by jbrestel** [comment]
it is not always from zero.   for example in the second intron from the left there is a line starting at high coverage and going down.  that is more of the issue i think because it appears like there is coverage when there isn't

**Comment by cmdcolin** [comment]
yes. there is likely some fudamentally funky things that this plugin does, I believe that it is also challenged by particular bigwig encodings, specifically "bedgraph" style encodings. Do you know how your bigwig files were prepared? And do you have any sample files I can look at?

I think there are three main "types" of bigwigs: bedgraph, variablestep, and fixedstep, and I think the bedgraph encoding is the main culprit for these rendering issues, but it could be a combo of bedgraph and there being gaps/sparsity in the data files.
